.travis.yml

```diff
@@ -0,0 +1,101 @@
+        - sudo ln -s ../../bin/ccache arm-none-eabi-g++
+        - export PATH="$PATH:/usr/lib/ccache"
+        - popd
+        # Install arm-none-eabi-
+        - pushd /home/travis/build && mkdir gcc-arm && cd gcc-arm
+        - curl -L0 "https://developer.arm.com/-/media/Files/downloads/gnu-rm/9-2020q2/gcc-arm-none-eabi-9-2020-q2-update-x86_64-linux.tar.bz2?revision=05382cca-1721-44e1-ae19-1e7c3dc96118&la=en&hash=D7C9D18FCA2DD9F894FD9F3C3DC9228498FA281A" --output gcc-arm-none-eabi-9-2020-q2-update.tar.bz2
+        - tar xf gcc-arm-none-eabi-9-2020-q2-update.tar.bz2
+        - export PATH="$PATH:${PWD}/gcc-arm-none-eabi-9-2020-q2-update/bin"
+        - popd
+        - arm-none-eabi-gcc --version
+        - pip install --upgrade cmake
+        # Install the changes added in the pull request
+        - pip install .
+        - mbedtools --version
+        # We use manual clone, with depth and single branch = the fastest
+        - cd ..
+        - git clone --depth=1 --single-branch --branch feature-cmake https://github.com/ARMmbed/${EXAMPLE_NAME}.git
+        - cd ${EXAMPLE_NAME}
+        - git clone --depth=1 --single-branch --branch feature-cmake https://github.com/ARMmbed/mbed-os.git
+      script:
+        - mbedtools configure -t GCC_ARM -m ${TARGET_NAME};
+        - mkdir -p build
+        - cd build && cmake .. -GNinja -DCMAKE_BUILD_TYPE=${PROFILE} && cmake --build .
+        - ccache -s
+
+    - <<: *cmake-build-test
+      name: "CMake blinky example - release (K64F)"
+      env: NAME=cmake_test EXAMPLE_NAME=mbed-os-example-blinky TARGET_NAME=K64F PROFILE=release
+
+    - <<: *cmake-build-test
+      name: "CMake blinky example - debug (K64F)"
+      env: NAME=cmake_test EXAMPLE_NAME=mbed-os-example-blinky TARGET_NAME=K64F PROFILE=debug
+
+    - <<: *cmake-build-test
+      name: "CMake blinky example - develop (DISCO_L475VG_IOT01A)"
+      env: NAME=cmake_test EXAMPLE_NAME=mbed-os-example-blinky TARGET_NAME=DISCO_L475VG_IOT01A PROFILE=develop
+
+    - <<: *cmake-build-test
+      name: "CMake blinky example - release (DISCO_L475VG_IOT01A)"
+      env: NAME=cmake_test EXAMPLE_NAME=mbed-os-example-blinky TARGET_NAME=DISCO_L475VG_IOT01A PROFILE=release
+
+    - <<: *cmake-build-test
+      name: "CMake blinky example - debug (DISCO_L475VG_IOT01A)"
+      env: NAME=cmake_test EXAMPLE_NAME=mbed-os-example-blinky TARGET_NAME=DISCO_L475VG_IOT01A PROFILE=debug
+
+    - <<: *cmake-build-test
+      name: "CMake blinky example - develop (NRF52840_DK)"
+      env: NAME=cmake_test EXAMPLE_NAME=mbed-os-example-blinky TARGET_NAME=NRF52840_DK PROFILE=develop
+
+    - <<: *cmake-build-test
+      name: "CMake blinky example - release (NRF52840_DK)"
+      env: NAME=cmake_test EXAMPLE_NAME=mbed-os-example-blinky TARGET_NAME=NRF52840_DK PROFILE=release
+
+    - <<: *cmake-build-test
+      name: "CMake blinky example - debug (NRF52840_DK)"
+      env: NAME=cmake_test EXAMPLE_NAME=mbed-os-example-blinky TARGET_NAME=NRF52840_DK PROFILE=debug
```

Comment:
Do we need to mention all the configs separately?  Can we have a loop which reads different targets and profiles instead?

Reply:
I prefer this to a loop. cmake_test is parameterized and each of these is just calling it. This is flat and easy to understand. When we have hundreds of targets, this will be very long: it may make sense to have a tool to help auto-generate this file. When we are rapidly adding targets, to decrease the maintenance burden of updating each of our travis.yml (in mbed-os, blinky, and also in tools as here), it may make sense to make the job smarter to detect which targets are supported and build each profile for them. For now, this is good enough, IMO.

Reply:
FWIW I prefer this flat, declarative structure too. CI scripts shouldn't have anything clever in them ;)

Reply:
We could use a loop to build all profiles, as they are pretty much a standard, if possible. Thanks.

Reply:
There doesn't seems to be a looping construct in Travis configuration https://docs.travis-ci.com/user/speeding-up-the-build to parallelise builds. As @Patater recommended, in future we can think about auto-generating `.travis.yml` file. But for time being, I'll leave this as it is.